repository/src/main/java/org/alfresco/repo/security/authentication/ResetPasswordServiceImpl.java

```diff
@@ -517,27 +532,33 @@ protected String getConfirmResetPasswordEmailTemplate(ClientApp clientApp)
     /**
      * This method creates a URL for the 'reset password' link which appears in the email
      */
-    protected String createResetPasswordUrl(ClientApp clientApp, final String id, final String key)
+    protected String createResetPasswordUrl(ClientApp clientApp, final String id, final String key, final String userName)
     {
         StringBuilder sb = new StringBuilder(100);
 
         String pageUrl = clientApp.getProperty("resetPasswordPageUrl");
-        if (StringUtils.isEmpty(pageUrl))
-        {
-            sb.append(UrlUtil.getShareUrl(sysAdminParams));
 
-            LOGGER.warn("'resetPasswordPageUrl' property is not set for the client [" + clientApp.getName()
-                        + "]. The default base url of Share will be used [" + sb.toString() + "]");
-        }
-        else
-        {
-            // We pass an empty string as we know that the pageUrl is not null
-            sb.append(getUrl(pageUrl, ""));
+        if(clientApp.getName().equals("adw")) {
```

Comment:
Apart from missing Alfresco code standard format, we shouldn't check for the specific registered-client. The client name is provided by the REST API caller, the API makes sure that the client exists, therefore, we should get all the required info from `ClientApp` object.